src/sumBy.ts

```diff
@@ -40,18 +58,34 @@ export function sumBy<T>(
   callbackfn: (value: T, index: number, data: ReadonlyArray<T>) => number,
 ): number;
 
+export function sumBy<T>(
+  data: ReadonlyArray<T>,
+  callbackfn: (value: T, index: number, data: ReadonlyArray<T>) => bigint,
+): bigint | 0;
+
 export function sumBy(...args: ReadonlyArray<unknown>): unknown {
   return purry(sumByImplementation, args);
 }
 
 const sumByImplementation = <T>(
   array: ReadonlyArray<T>,
-  callbackfn: (value: T, index: number, data: ReadonlyArray<T>) => number,
-): number => {
-  let sum = 0;
+  callbackfn: (
+    value: T,
+    index: number,
+    data: ReadonlyArray<T>,
+  ) => bigint | number,
+): bigint | number => {
+  let sum: bigint | number | undefined;
   for (const [index, item] of array.entries()) {
     const summand = callbackfn(item, index, array);
-    sum += summand;
+    if (sum === undefined) {
```

Comment:
This adds a check on every iteration, but we know it's only relevant on the first iteration. We should run the mapper on the first item before the loop and use it to initialize `sum`.

Reply:
thx for the suggestion. I've made a change to avoid checking for the undefined in the loop